face/capture/src/main/java/com/simprints/face/capture/screens/FaceCaptureViewModel.kt

```diff
@@ -77,24 +83,51 @@ internal class FaceCaptureViewModel @Inject constructor(
     }
 
     fun initFaceBioSdk(activity: Activity) = viewModelScope.launch {
-        val license = licenseRepository.getCachedLicense(Vendor.RANK_ONE)
+        val licenseVendor = Vendor.RANK_ONE
+        var license = licenseRepository.getCachedLicense(licenseVendor)
         var licenseStatus = license.determineLicenseStatus()
 
         if (licenseStatus == LicenseStatus.VALID) {
-            val initializer = resolveFaceBioSdk().initializer
-            if (!initializer.tryInitWithLicense(activity, license!!.data)) {
-                // License is valid but the SDK failed to initialize
-                // This is should reported as an error
-                licenseStatus = LicenseStatus.ERROR
+            licenseStatus = initialize(activity, license!!)
+        }
+        // In some cases license is invalidated on initialisation attempt
+        if (licenseStatus != LicenseStatus.VALID) {
```

Comment:
@luhmirin-s It seems like there's quite a bit of code inside the `initFaceBioSdk` function. Please consider simplifying or restructuring it to make it a bit lighter.

Reply:
Extracted licence refresh into a method.